src/core/resolvers/mergeAllof.ts

```diff
@@ -0,0 +1,33 @@
+import { OpenAPIParser } from 'redoc';
+import { OpenAPISchema } from 'redoc/typings/types';
+
+export const mergeAllOf = (
+  parser: OpenAPIParser,
+  inputSchema: OpenAPISchema,
+) => {
+  let schema = { ...inputSchema };
+  // test case [TopLevelAllOf]
+  if (schema?.allOf) {
+    schema = { ...schema, ...parser.mergeAllOf(schema) };
```

Comment:
Hey @yokoyama-takuya. We use allOf in our schemas in combination with oneOf that reference each other.
I tried your pr and I had to make a small change to make this use case work. Not sure if the same is also needed for nestedAllOf

Before:
```typescript
 schema = { ...schema, ...parser.mergeAllOf(schema) };
```

after
```typescript
    const oneOf = schema.oneOf;
    schema = {
      ...schema,
      ...parser.mergeAllOf(schema),
    };
    schema.oneOf = oneOf;
```
Short example:
```
Pet:
  oneOf:
   - Cat
   - Dog
  Properties:
   - id
   - name
Cat: 
  allOf:
   - Pet
Dog: 
  allOf:
   - Pet
```

Reply:
Hi! @koenvangeert  Thanks to try this!
If it's okay until next week , I'll make an additional commit to this PR.